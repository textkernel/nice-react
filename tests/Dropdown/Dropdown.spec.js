import React from 'react';
import { Dropdown } from '../../dist';

describe('<Dropdown />', () => {
    it('should render a simple Dropdown', () => {
        const wrapper = shallow(<Dropdown label={ 'Nice dropdown' } items={ [{
            label: 'This is a dropdown heading',
            heading: true
        }, {
            label: 'This is a short description for this dropdown',
            description: true
        }, {
            label: 'First',
            value: 'first',
            icon: 'fa-download'
        }, {
            label: 'Second',
            context: 'bad',
            icon: 'fa-share'
        }, {
            separator: true
        }, {
            label: 'Third',
            disabled: true,
            icon: 'fa-bookmark'
        }] }
    />);
        expect(wrapper).toMatchSnapshot();
    });

    it('should render Dropdown with different context and alignment', () => {
        const wrapper = shallow(<Dropdown label={ 'Primary' } items={ [{
            label: 'First'
        }, {
            label: 'Second'
        }, {
            label: 'Third'
        }] } context="primary" dropUp alignLeft
    />);
        expect(wrapper).toMatchSnapshot();
    });

    it('should render with multiple selection', () => {
        const wrapper = shallow(<Dropdown label={ 'Multiselect' } items={ [{
            label: 'First',
            value: 'first'
        }, {
            label: 'Second',
            value: 'second'
        }, {
            label: 'Third',
            value: 'third'
        }] } multiple />);
        expect(wrapper).toMatchSnapshot();
    });

    it('should render with radios', () => {
        const wrapper = shallow(<Dropdown label={ 'Multiselect' } items={ [{
            label: 'First',
            value: 'first'
        }, {
            label: 'Second',
            value: 'second'
        }, {
            label: 'Third',
            value: 'third'
        }] } radios />);
        expect(wrapper).toMatchSnapshot();
    });

    it('should render Dropdown with custom classname/style', () => {
        const wrapper = shallow(
            <Dropdown 
                className="my-custom-classname"
                style={{ right: 0 }}
                label={ 'Primary' } 
                items={ [{ label: 'First' }, { label: 'Second' }, { label: 'Third' }] }
                context="primary"
                dropUp
                alignLeft 
            />
        );

        expect(wrapper).toMatchSnapshot();
    });
});
