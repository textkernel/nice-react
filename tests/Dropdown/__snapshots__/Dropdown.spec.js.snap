// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dropdown /> should render Dropdown with different context and alignment 1`] = `
<div
  className="nice-dropdown primary medium drop-up align-left"
  onMouseOut={[Function]}
  onMouseOver={[Function]}
>
  <Button
    alternative={false}
    context="primary"
    href={null}
    icon={null}
    onClick={[Function]}
    pullRight={false}
    size="medium"
    submit={false}
  >
    Primary
     
    <i
      className="icon fa-angle-down"
    />
  </Button>
  <div
    className="dropdown-menu"
  >
    <li
      className=""
      key="0"
    >
      <a
        href="#"
        onClick={[Function]}
      >
        First
      </a>
    </li>
    <li
      className=""
      key="1"
    >
      <a
        href="#"
        onClick={[Function]}
      >
        Second
      </a>
    </li>
    <li
      className=""
      key="2"
    >
      <a
        href="#"
        onClick={[Function]}
      >
        Third
      </a>
    </li>
  </div>
</div>
`;

exports[`<Dropdown /> should render a simple Dropdown 1`] = `
<div
  className="nice-dropdown primary medium"
  onMouseOut={[Function]}
  onMouseOver={[Function]}
>
  <Button
    alternative={false}
    context="primary"
    href={null}
    icon={null}
    onClick={[Function]}
    pullRight={false}
    size="medium"
    submit={false}
  >
    Nice dropdown
     
    <i
      className="icon fa-angle-down"
    />
  </Button>
  <div
    className="dropdown-menu"
  >
    <li
      className="heading"
      key="0"
    >
      <span>
        This is a dropdown heading
      </span>
    </li>
    <li
      className="description"
      key="1"
    >
      <span>
        This is a short description for this dropdown
      </span>
    </li>
    <li
      className=""
      key="2"
    >
      <a
        href="#"
        onClick={[Function]}
      >
        <i
          className="icon fa-download"
        />
        First
      </a>
    </li>
    <li
      className="bad"
      key="3"
    >
      <a
        href="#"
        onClick={[Function]}
      >
        <i
          className="icon fa-share"
        />
        Second
      </a>
    </li>
    <li
      className="separator"
      key="4"
    />
    <li
      className="disabled"
      key="5"
    >
      <a
        href="#"
        onClick={[Function]}
      >
        <i
          className="icon fa-bookmark"
        />
        Third
      </a>
    </li>
  </div>
</div>
`;

exports[`<Dropdown /> should render with multiple selection 1`] = `
<div
  className="nice-dropdown primary medium"
  onMouseOut={[Function]}
  onMouseOver={[Function]}
>
  <Button
    alternative={false}
    context="primary"
    href={null}
    icon={null}
    onClick={[Function]}
    pullRight={false}
    size="medium"
    submit={false}
  >
    Multiselect
     
    <i
      className="icon fa-angle-down"
    />
  </Button>
  <div
    className="dropdown-menu"
  >
    <li
      className=""
      key="0"
    >
      <label>
        <input
          checked={false}
          onChange={[Function]}
          type="checkbox"
        />
        First
      </label>
    </li>
    <li
      className=""
      key="1"
    >
      <label>
        <input
          checked={false}
          onChange={[Function]}
          type="checkbox"
        />
        Second
      </label>
    </li>
    <li
      className=""
      key="2"
    >
      <label>
        <input
          checked={false}
          onChange={[Function]}
          type="checkbox"
        />
        Third
      </label>
    </li>
  </div>
</div>
`;

exports[`<Dropdown /> should render with radios 1`] = `
<div
  className="nice-dropdown primary medium"
  onMouseOut={[Function]}
  onMouseOver={[Function]}
>
  <Button
    alternative={false}
    context="primary"
    href={null}
    icon={null}
    onClick={[Function]}
    pullRight={false}
    size="medium"
    submit={false}
  >
    Multiselect
     
    <i
      className="icon fa-angle-down"
    />
  </Button>
  <div
    className="dropdown-menu"
  >
    <li
      className=""
      key="0"
    >
      <label>
        <input
          checked={true}
          onChange={[Function]}
          type="radio"
        />
        First
      </label>
    </li>
    <li
      className=""
      key="1"
    >
      <label>
        <input
          checked={false}
          onChange={[Function]}
          type="radio"
        />
        Second
      </label>
    </li>
    <li
      className=""
      key="2"
    >
      <label>
        <input
          checked={false}
          onChange={[Function]}
          type="radio"
        />
        Third
      </label>
    </li>
  </div>
</div>
`;
